src/distributed_graph_generator.cc

```diff
@@ -20,19 +20,37 @@ distributed_graph_generator::distributed_graph_generator(
 	// Build initial epoch command (this is required to properly handle anti-dependencies on host-initialized buffers).
 	// We manually generate the first command, this will be replaced by applied horizons or explicit epochs down the line (see
 	// set_epoch_for_new_commands).
-	auto* const epoch_cmd = cdag.create<epoch_command>(task_manager::initial_epoch_task, epoch_action::none);
+	const auto epoch_tsk = tm.get_task(task_manager::initial_epoch_task);
+	auto* const epoch_cmd = cdag.create<epoch_command>(epoch_tsk->get_id(), epoch_action::none);
 	epoch_cmd->mark_as_flushed(); // there is no point in flushing the initial epoch command
-	if(m_recorder != nullptr) m_recorder->record_command(*epoch_cmd);
+	if(m_recorder != nullptr) { *m_recorder << command_record(*epoch_cmd, *epoch_tsk, {}); }
```

Comment:
The new design retrieves the task from the task manager here even if recording is disabled (not that it matters in this context, but it's unnecessary).

Reply:
Fixed.